controllers/components/datasciencepipelines/datasciencepipelines_controller_actions.go

```diff
@@ -0,0 +1,103 @@
+
+	componentsv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/controllers/status"
+	odhtypes "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
+	odhdeploy "github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
+)
+
+func initialize(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	rr.Manifests = append(rr.Manifests, odhtypes.ManifestInfo{
+		Path:       DefaultPath,
+		ContextDir: "",
+		SourcePath: "",
+	})
+	if err := odhdeploy.ApplyParams(DefaultPath, nil, map[string]string{"namespace": rr.DSCI.Spec.ApplicationsNamespace}); err != nil {
+		return fmt.Errorf("failed to update params.env from %s : %w", rr.Manifests[0], err)
+	}
+	return nil
+}
+
+func devFlags(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	dsp, ok := rr.Instance.(*componentsv1.DataSciencePipelines)
+	if !ok {
+		return fmt.Errorf("resource instance %v is not a componentsv1.DataSciencePipelines)", rr.Instance)
+	}
+
+	if dsp.Spec.DevFlags == nil {
+		return nil
+	}
+
+	// Implement devflags support logic
+	// If dev flags are set, update default manifests path
+	if len(dsp.Spec.DevFlags.Manifests) != 0 {
+		manifestConfig := dsp.Spec.DevFlags.Manifests[0]
+		if err := odhdeploy.DownloadManifests(ctx, componentsv1.DataSciencePipelinesComponentName, manifestConfig); err != nil {
+			return err
+		}
+
+		if manifestConfig.SourcePath != "" {
+			rr.Manifests[0].Path = odhdeploy.DefaultManifestPath
+			rr.Manifests[0].ContextDir = componentsv1.DataSciencePipelinesComponentName
+			rr.Manifests[0].SourcePath = manifestConfig.SourcePath
+		}
+	}
+
+	return nil
+}
+
+func UnmanagedArgoWorkFlowExists(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	// Check preconditions if this is an upgrade
+	if rr.Instance.GetStatus().Phase != status.PhaseReady {
+		return nil
+	}
+
+	workflowCRD := &apiextensionsv1.CustomResourceDefinition{}
+	if err := rr.Client.Get(ctx, client.ObjectKey{Name: ArgoWorkflowCRD}, workflowCRD); err != nil {
+		if k8serr.IsNotFound(err) {
+			return nil
+		}
+		return fmt.Errorf("failed to get existing Workflow CRD : %w", err)
+	}
+
+	// Verify if existing workflow is deployed by ODH with label
+	// if not then set Argo capability condition to false
+	odhLabelValue, odhLabelExists := workflowCRD.Labels[labels.ODH.Component(componentsv1.DataSciencePipelinesComponentName)]
+	if !odhLabelExists || odhLabelValue != "true" {
+		message := fmt.Sprintf("Failed upgrade: %s CRD already exists but not deployed by this operator"+
+			"Remove existing Argo workflows or set `spec.components.datasciencepipelines.managementState` to Removed to proceed", ArgoWorkflowCRD)
+
+		status.SetCondition(&rr.DSC.Status.Conditions, string(status.CapabilityDSPv2Argo), status.ArgoWorkflowExist, message, corev1.ConditionFalse)
```

Comment:
I don't think the component should change the status of the DSC (in fact, there is not guarantee that the status of the DS than included this information) but it should only amend its own conditions. If this is a precondition, then it should probably do something like https://github.com/opendatahub-io/opendatahub-operator/blob/feature-operator-refactor/controllers/components/modelregistry/modelregistry_controller_actions.go#L26-L48

Reply:
Reporting the status up to the DSC if needed, then should be part of https://issues.redhat.com/browse/RHOAIENG-13314 (I guess)

Reply:
Updated it now